server/src/building/BaseCommand.cpp

```diff
@@ -20,26 +20,36 @@ namespace utbot {
     BaseCommand::BaseCommand(std::list<std::string> commandLine, fs::path directory, bool shouldChangeDirectory)
         : commandLine(std::move(commandLine)), directory(std::move(directory)), shouldChangeDirectory{shouldChangeDirectory} {
         initOptimizationLevel();
+        initCompiler();
+        initOutput();
     }
     BaseCommand::BaseCommand(std::vector<std::string> commandLine, fs::path directory, bool shouldChangeDirectory)
         : commandLine(commandLine.begin(), commandLine.end()), directory(std::move(directory)), shouldChangeDirectory{shouldChangeDirectory} {
         initOptimizationLevel();
+        initCompiler();
+        initOutput();
     }
 
     BaseCommand::BaseCommand(BaseCommand const &other)
-        : directory(other.directory), commandLine(other.commandLine),
-          environmentVariables(other.environmentVariables), shouldChangeDirectory(other.shouldChangeDirectory) {
+            : directory(other.directory), commandLine(other.commandLine),
+              environmentVariables(other.environmentVariables), shouldChangeDirectory(other.shouldChangeDirectory),
+              compiler(other.compiler),
+              output(other.output) {
         if (other.optimizationLevel.has_value()) {
             optimizationLevel =
-                std::next(commandLine.begin(),
-                          std::distance<const_iterator>(other.commandLine.begin(),
-                                                        other.optimizationLevel.value()));
+                    std::next(commandLine.begin(),
+                              std::distance<const_iterator>(other.commandLine.begin(),
+                                                            other.optimizationLevel.value()));
         }
```

Comment:
from ld man
```
-O level
If level is a numeric values greater than zero ld optimizes the output. This might take significantly longer and therefore probably should only be enabled for the final binary. At the moment this option only affects ELF shared library generation. Future releases of the linker may make more use of this option. Also currently there is no difference in the linker's behaviour for different non-zero values of this option. Again this may change with future releases.
```

Reply:
Quite interesting to know 👍 Please make sure that appropriate value is set at different stages of the generation